[bug] omnictl error during config merge

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

```
omnictl config merge ~/.config/omni/config/omniconfig.yaml
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe441fb]

goroutine 1 [running]:
github.com/siderolabs/omni/client/pkg/omnictl/config.(*Config).Merge(0x0, {0x7ffd1db8d6e2?, 0x1?})
	/src/client/pkg/omnictl/config/config.go:164 +0x19b
github.com/siderolabs/omni/client/pkg/omnictl.init.func8(0xc000418d00?, {0xc0003fa7b0, 0x1, 0x141b93c?})
	/src/client/pkg/omnictl/config.go:193 +0x74
github.com/spf13/cobra.(*Command).execute(0x21bd620, {0xc0003fa780, 0x1, 0x1})
	/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0x21bed20)
	/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(0x142d702?)
	/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041 +0x13
main.main()
	/src/cmd/omnictl/main.go:25 +0x171
```

### Expected Behavior

No error

### Steps To Reproduce

```
omnictl -v
omnictl version v0.42.3 (API Version: 1)
```
linux-amd64

### What browsers are you seeing the problem on?

_No response_

### Anything else?

_No response_

--- a/client/pkg/omnictl/config/config.go
+++ b/client/pkg/omnictl/config/config.go
@@ -42,19 +42,22 @@ var (
 func Init(path string, create bool) (*Config, error) {
 	conf, err := load(path)
 
-	if os.IsNotExist(err) {
+	switch {
+	case os.IsNotExist(err):
 		if !create {
 			return nil, err
 		}
 
 		defaultConfig.Path = path
 
-		err := defaultConfig.Save()
+		err = defaultConfig.Save()
 		if err != nil {
 			return nil, err
 		}
 
 		conf = &defaultConfig
+	case err != nil:
+		return nil, err
 	}
 
 	current = conf

